common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -571,6 +799,76 @@ public AccountRecord getAccountWithLocalAccountId(@Nullable final String environ
         return Collections.unmodifiableList(accountsForThisApp);
     }
 
+    @Override
+    public List<AccountRecord> getAllTenantAccountsForAccountByClientId(@NonNull final String clientId,
+                                                                        @NonNull final AccountRecord accountRecord) {
+        final List<AccountRecord> allTenantAccounts = new ArrayList<>();
+
+        // Add the supplied AccountRecord as the 0th element...
+        allTenantAccounts.add(accountRecord);
+
+        // Grab all the accounts which might match
+        final List<AccountRecord> allMatchingAccountsByHomeId =
+                mAccountCredentialCache.getAccountsFilteredBy(
+                        accountRecord.getHomeAccountId(),
+                        accountRecord.getEnvironment(),
+                        null // realm
+                );
+
+        // Grab all of the AccountRecords associated with this clientId
+        final List<AccountRecord> allAppAccounts = getAccounts(
+                accountRecord.getEnvironment(),
+                clientId
+        );
+
+        // Iterate and populate
+        for (final AccountRecord acct : allAppAccounts) {
+            if (allMatchingAccountsByHomeId.contains(acct) && !accountRecord.equals(acct)) {
```

Comment:
Did we override `AccountRecord` equals?

Reply:
Yup - [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/dto/AccountRecord.java#L426-L477).